packages/@aws-cdk/aws-events/lib/connection.ts

```diff
@@ -0,0 +1,423 @@
+   * @default - None
+   */
+  readonly apiKeyAuthParameters?: ApiKeyAuthParameters;
+
+  /**
+   * Contains the Basic authorization parameters to use for the connection.
+   * @default - None
+   */
+  readonly basicAuthParameters?: BasicAuthParameters;
+
+  /**
+   * Contains the API key authorization parameters to use for the connection.
+   * @default - None
+   */
+  readonly invocationHttpParameters?: HttpParameters;
+
+  /**
+   * Contains the OAuth authorization parameters to use for the connection.
+   * @default - None
+   */
+  readonly oAuthParameters?: OAuthParameters;
+}
+
+/**
+ * The event connection base properties
+ */
+export interface BaseConnectionProps {
+  /**
+   * The name of the connection.
+   *
+   * @default - none
+   */
+  readonly authorizationType: AuthorizationType;
+
+  /**
+   * The name of the connection.
+   *
+   * @default - none
+   */
+  readonly authParameters: AuthParameters;
+
+  /**
+   * The name of the connection.
+   *
+   * @default - none
+   */
+  readonly description: string;
+
+  /**
+   * The name of the connection.
+   *
+   * @default - none
+   */
+  readonly connectionName: string;
+}
+
+
+/**
+ * The event connection properties
+ */
+export interface ConnectionProps extends BaseConnectionProps {
+
+}
+
+/**
+ * Interface for EventBus Connections
+ */
+export interface IConnection extends IResource {
+  /**
+   * The Name for the connection.
+   * @attribute
+   */
+  readonly connectionName: string;
+
+  /**
+   * The ARN of the connection created.
+   * @attribute
+   */
+  readonly connectionArn: string;
+
+  /**
+   * The ARN for the secret created for the connection.
+   * @attribute
+   */
+  readonly connectionSecretArn: string;
+}
+
+/**
+ * The event connection properties
+ */
+export interface ConnectionProps extends BaseConnectionProps {
+
+}
+
+/**
+ * Interface with properties necessary to import a reusable Connection
+ */
+export interface ConnectionAttributes {
+  /**
+   * The Name for the connection.
+   */
+  readonly connectionName: string;
+
+  /**
+   * The ARN of the connection created.
+   */
+  readonly connectionArn: string;
+
+  /**
+   * The ARN for the secret created for the connection.
+   */
+  readonly connectionSecretArn: string;
+}
+
+/**
+ * Define an EventBridge Connection
+ *
+ * @resource AWS::Events::Connection
+ */
+export class Connection extends Resource implements IConnection {
+  /**
+   * Import an existing connection resource
+   * @param scope Parent construct
+   * @param id Construct ID
+   * @param connectionArn ARN of imported connection
+   */
+  public static fromEventBusArn(scope: Construct, id: string, connectionArn: string, connectionSecretArn: string): IConnection {
+    const parts = Stack.of(scope).parseArn(connectionArn);
+
+    return new ImportedConnection(scope, id, {
+      connectionArn: connectionArn,
+      connectionName: parts.resourceName || '',
```

Comment:
I think we should consider the `arn` invalid if `parts.resourceName` does not exist.

```ts
if (!parts.resourceName) {
      throw new Error('invalid events connection ARN');
}
```